plugins/filter/resource_list_compare.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: resource_list_compare
+    author: Rohit Thakur (@rohitthakur2590)
+    version_added: "2.4.0"
+    short_description: Generate the final resource list combining/comparing base and provided resources
+    description:
+        - Generate the final list of resources after comparing with base list and provide list of resources/bangs.
+    options:
+      base:
+        description: Specify the base list of supported network resources.
+        type: list
+        elements: str
+      target:
+        description: Specify the target list of resources.
+        type: list
+        elements: str
+"""
+
+EXAMPLES = r"""
+- name: set facts for data and criteria
+  set_fact:
+    network_resources: "{{ network_resources_list['modules']|resource_list_compare(resources) }}"
+"""
+
+RETURN = """
+  list:
+    combined=['interfaces', 'l2_interfaces', 'l3_interfaces']
```

Comment:
return value shouldn't have values, but instead a description of what's returned

Reply:
Good find. Thanks!